### PR TITLE
examples: Add bmi270 sample application without SCU

### PR DESCRIPTION
**Pull request by shino-ixy**
I've written User Mode Driver for BMI270(without SCU), and create sample application that can visible tilt of SPRESENSE Main Board by LED[0-3].

### CODE REVIEW

File: sdk/configs/examples/bmi270_no_scu/defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
++CXD56_I2C0_SCUSEQ=n
```
Comment (SPRESENSE) [inline sdk/configs/examples/bmi270_no_scu/defconfig:1]: Create a `defconfig` using `mkdefconfig.py`.
```
$ cd spresense/sdk
$ ./tools/mkdefconfig.py examples/bmi270_no_scu
```

File: examples/bmi270_no_scu/Kconfig [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+config EXAMPLES_BMI270_NO_SCU_PRIORITY
+	int "Sensor task priority"
+	default 100
+
+config EXAMPLES_BMI270_NO_SCU_STACKSIZE
+	int "Sensor stack size"
+	default 2048
+
+config EXAMPLES_BMI270_NO_SCU_DEBUFON
```
Comment (SPRESENSE) [inline examples/bmi270_no_scu/Kconfig:29]:
```suggestion
config EXAMPLES_BMI270_NO_SCU_DEBUGON
```

File: examples/bmi270_no_scu/Kconfig [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	tristate "Enable Debug Print"
```
Comment (SPRESENSE) [inline examples/bmi270_no_scu/Kconfig:30]:
```suggestion
	bool "Enable Debug Print"
```

**Review by SPRESENSE — CHANGES_REQUESTED** [review]
@shino-ixy Thank you for your contribution!

* Please change the target branch from `master` to `develop`.
(See https://github.com/sonydevworld/spresense/blob/master/CONTRIBUTING.md)

* Please remove some `.DS_Store` files.
I will add this filename into `.gitignore`.

* commit message
The prefix before the colon `:` indicates the context.
Referring to other commits, change it and PR title as below.
"examples: Add bmi270 sample application without SCU"

* coding style
Please change your coding style to match the NuttX's nxstyle. You can use the `checkpatch.sh` tool. e.g.
```
$ cd spresense/sdk
$ ../nuttx/tools/checkpatch.sh -f <filename>
```

* License
 * For files without a license description, please attach an Apache 2.0 license.
(Refer to https://github.com/sonydevworld/spresense-nuttx-apps/blob/new-master/examples/scd41/scd41_main.c#L1-#L19)
 * It seems an `i2c_bmi270.c` refers to the original Bosch license.
Please be sure to grant the appropriate license.
If you have mixed licenses, please separate files for each license.

Let me review the rest after the revised pull-request is issued.

File: sdk/configs/examples/bmi270_no_scu/defconfig (hunk 1 above)
Comment (shino-ixy) [reply sdk/configs/examples/bmi270_no_scu/defconfig]: Thanks, teaching usage of tool. I success to generate defconfig

File: examples/bmi270_no_scu/Kconfig (hunk 2 above)
Comment (shino-ixy) [reply examples/bmi270_no_scu/Kconfig]: as you said. Thank you.

File: examples/bmi270_no_scu/Kconfig (hunk 3 above)
Comment (shino-ixy) [reply examples/bmi270_no_scu/Kconfig]: as you said. Thank you.

**Comment by shino-ixy** [comment]
@SPRESENSE 
Thank you for Reviewing!!

> Please change the target branch from master to develop.
> commit message
> 
OK, I'd changed.

> Please remove some .DS_Store files.
Sorry, I'd deleted these files.

> coding style
Thank you for attention. 
I've updated each codes.

> License
I've separeted codes of BMI270 config table for separating License,
and insert Licesnse each files.

Please re-Review this pull request.

**Review by SPRESENSE — CHANGES_REQUESTED** [review]
Let me suggest changing the application name from `bmi270_no_scu` to `i2c_direct`.

As you know, the bmi270 driver should originally be created as a sensor driver.
However, it is also possible to control `/dev/i2c` directly, as this example application.

I think it would be appropriate to use the generic name `i2c_direct` as a sample for controlling an i2c character device. Initially only the bmi270 will exist, but we would like to be able to add devices in the future.

And,  please rebase as it contains a merge commit.

**Comment by shino-ixy** [comment]
***@***.***> @sonydevworld/spresense 

Thank you for reviewing.

> Let me suggest changing the application name from bmi270_no_scu to i2c_direct.

That’s right! I’ve renamed files and definitions in codes.

And I’ve updated codes to follow each comment, applied rabasing “develop”.

File: examples/i2c_direct/include/i2c_bmi270.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+void fini_bmi270(i2c_bmi270_t *pctrl);
+int exec_dequeue_fifo(i2c_bmi270_t *pctrl);
+int get_latest_acc(axis_t *pd, i2c_bmi270_t *pctrl);
+int get_latest_gyr(axis_t *pd, i2c_bmi270_t *pctrl);
+
+/* bmi270.c */
+
+int get_bmi270_config_file_size(void);
+uint8_t *get_bmi270_config_file_addr(void);
+
```
Comment (SPRESENSE) [inline examples/i2c_direct/include/i2c_bmi270.h:83]:
```suggestion

#if defined(__cplusplus)
}
#endif

```

File: examples/i2c_direct/include/i2c_common.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  int i2c_addr;
+  int speed;
+} i2c_ctrl_t;
+
+int i2c_reg_write(i2c_ctrl_t *pi2c, uint8_t reg, uint8_t value);
+int i2c_reg_write_burst(i2c_ctrl_t *pi2c,
+  uint8_t reg, uint8_t *pvalue, int len);
+int i2c_reg_read(i2c_ctrl_t *pi2c,
+  uint8_t reg, uint8_t *value, int16_t len);
+
```
Comment (SPRESENSE) [inline examples/i2c_direct/include/i2c_common.h:54]:
```suggestion

#if defined(__cplusplus)
}
#endif

```

File: examples/i2c_direct/i2c_direct_main.c [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+{
+  board_gpio_write(id, false);
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/**
+ * @brief main - init BMI270 and mesurement 100 times.
```
Comment (SPRESENSE) [inline examples/i2c_direct/i2c_direct_main.c:149]:
```suggestion
 * @brief main - init BMI270 and measurement 512 times.
```

File: examples/i2c_direct/i2c_direct_main.c [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+#define LED0  PIN_I2S1_DATA_OUT
+#define LED1  PIN_I2S1_DATA_IN
+#define LED2  PIN_I2S1_LRCK
+#define LED3  PIN_I2S1_BCK 
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static void gpio_setup_direction (int id, bool input);
```
Comment (SPRESENSE) [inline examples/i2c_direct/i2c_direct_main.c:60]:
```suggestion
static void gpio_setup_direction(int id, bool input);
```

**Comment by SPRESENSE** [comment]
@shino-ixy  Remove some warnings.
```
i2c_bmi270.c: In function 'bmi270_fifo_decoder':
i2c_bmi270.c:244:10: warning: 'return' with a value, in function returning void [-Wreturn-type]
  244 |   return 0;
      |          ^
i2c_bmi270.c:161:1: note: declared here
  161 | bmi270_fifo_decoder
      | ^~~~~~~~~~~~~~~~~~~
i2c_bmi270.c: In function 'init_bmi270':
i2c_bmi270.c:312:20: warning: assignment to 'axis_t *' {aka 'struct _axis_type *'} from incompatible pointer type 'uint8_t *' {aka 'unsigned char *'} [-Wincompatible-pointer-types]
  312 |   pctrl->acc_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
      |                    ^
i2c_bmi270.c:322:20: warning: assignment to 'axis_t *' {aka 'struct _axis_type *'} from incompatible pointer type 'uint8_t *' {aka 'unsigned char *'} [-Wincompatible-pointer-types]
  322 |   pctrl->gyr_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
      |                    ^
i2c_bmi270.c: At top level:
i2c_bmi270.c:108:15: warning: array 'bmi270_config_file' assumed to have one element
  108 | const uint8_t bmi270_config_file[];
      |               ^~~~~~~~~~~~~~~~~~
bmi270.c: In function 'get_bmi270_config_file_addr':
bmi270.c:947:10: warning: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  947 |   return bmi270_config_file;
      |          ^~~~~~~~~~~~~~~~~~
```

**Comment by shino-ixy** [comment]
I’m so sorry for repeated reviewing.

I've updated comments and codes, resolved any issues. these updates are included in single commit.

File: examples/i2c_direct/include/i2c_bmi270.h [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+#include <sys/ioctl.h>
+#include <nuttx/i2c/i2c_master.h>
+#include <nuttx/arch.h>
+#include <arch/board/board.h>
+#include <arch/chip/pin.h>
+#include "i2c_common.h"
+#include "debug_printf.h"
+
+/****************************************************************************
+ * Public Types
```
Comment (SPRESENSE) [inline examples/i2c_direct/include/i2c_bmi270.h:40]:
```suggestion
 * Pre-processor Definitions
```

**Review by SPRESENSE — CHANGES_REQUESTED** [review]
@shino-ixy , thanks for your quick response.
I added a comment and please fix `i2c_bmi270.h`.

**Comment by shino-ixy** [comment]
Thank you for checking, I've updated i2c_bmi270.h.

**Comment by SPRESENSE** [comment]
@shino-ixy san, could you please add sdk/configs/examples/i2c_direct/README.txt?
Thanks.

**Comment by shino-ixy** [comment]
OK! I've just written README.txt, Is this content sufficient?

**Comment by SPRESENSE** [comment]
@shino-ixy san
README.txt you added is very helpful. Thank you.

And, please refer to [README.txt](https://github.com/sonydevworld/spresense/blob/develop/sdk/configs/examples/alarm/README.txt), and add a simple README.txt under sdk/configs/examples/i2c_direct directory.

**Comment by shino-ixy** [comment]
@SPRESENSE Thank you for checking, I've just added README in configs.

File: sdk/configs/examples/i2c_direct/README.txt [hunk 9]
```diff
@@ -0,0 +1,4 @@
+This configuration contains required options to use gyro example.
```
Comment (SPRESENSE) [inline sdk/configs/examples/i2c_direct/README.txt:1]:
```suggestion
This configuration contains required options to use i2c_direct example.
```

File: sdk/configs/examples/i2c_direct/README.txt [hunk 10]
```diff
@@ -0,0 +1,4 @@
...
+
+[Source path]
+examples/gyro
```
Comment (SPRESENSE) [inline sdk/configs/examples/i2c_direct/README.txt:4]:
```suggestion
examples/i2c_direct
```

**Review by SPRESENSE — CHANGES_REQUESTED** [review]
@shino-ixy , please modify an example name as I suggested.

**Comment by shino-ixy** [comment]
I'm so sorry...for my wrong commit.

**Review by SPRESENSE — APPROVED** [review]
@shino-ixy san, thank you!!

**Pull request merged**